### ISSUE TITLE
<AnalysisLevel>latest-all</AnalysisLevel> seems to be not effective when switching to .NET7

### ISSUE THREAD

**Issue by moses70** [dotnet/roslyn#65630]
**Version Used**: 
SDK 7.0.100 with Visual studio 17.4.1

**Steps to Reproduce**:

1. Install .NET6 SDK and .NET 7 SDK in parallel (latest versions)
2. Create a C# project with a global.json selecting .NET6
3. Use &lt;AnalysisLevel&gt;latest-all&lt;/AnalysisLevel&gt; in csproj
4. Using some example code, targeting .NET48 with &lt;EnableNETAnalyzers&gt;true&lt;/EnableNETAnalyzers&gt;
5. Verify the number of code analysis warnings. E.g. 97 warnings
6. Switch to .NET7 using global.json, rebuild
7. The number of code analysis warnings decreases to about the same number as with &lt;AnalysisLevel&gt;latest-recommended&lt;/AnalysisLevel&gt;

**Diagnostic Id**:

For example, CA2007 should be enabled in "latest-all", but is not in .NET7

**Expected Behavior**:
In .NET7, at least the same number of code style warnings should be visible as in .NET 6.

**Actual Behavior**:
<AnalysisLevel>latest-all</AnalysisLevel> has not the same effect in .NET 7

**Workaround**
With this global.json, I can still use C#11 features (with VS17.4) and all analyzers are working:
{
  "sdk": {
    "allowPrerelease": false,
    "version": "6.0.0",
    "rollForward": "latestMinor"
  }
}

**Comment by Youssef1313** [comment]
@mavasani Is this the issue fixed in https://github.com/dotnet/roslyn-analyzers/pull/6258 ?

**Comment by mavasani** [comment]
> @mavasani Is this the issue fixed in [dotnet/roslyn-analyzers#6246](https://github.com/dotnet/roslyn-analyzers/pull/6246) ?

Seems very likely. @moses70 Can you please check if adding an explicit NuGet package reference to https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet7/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/versions fixes the issue? This fix has been inserted into the dotnet/sdk repo and should be part of the next .NET 7 servicing patch.

**Comment by moses70** [comment]
Works for me!
Thank you for the quick response and workaround.

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/6245
